applications/MeshingApplication/custom_processes/metrics_spr_error_process.cpp

```diff
@@ -0,0 +1,591 @@
+// KRATOS  __  __ _____ ____  _   _ ___ _   _  ____ 
+//        |  \/  | ____/ ___|| | | |_ _| \ | |/ ___|
+//        | |\/| |  _| \___ \| |_| || ||  \| | |  _ 
+//        | |  | | |___ ___) |  _  || || |\  | |_| |
+//        |_|  |_|_____|____/|_| |_|___|_| \_|\____| APPLICATION
+//
+//  License:		 BSD License
+//                       license: MeshingApplication/license.txt
+//
+//  Main authors:    Anna Rehr
+//  Co-author   :    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_processes/metrics_spr_error_process.h"
+#include "utilities/variable_utils.h"
+#include "linear_solvers/skyline_lu_factorization_solver.h"
+
+namespace Kratos
+{
+template<SizeType TDim>
+SPRMetricProcess<TDim>::SPRMetricProcess(
+    ModelPart& rThisModelPart,
+    Parameters ThisParameters,
+    LinearSolverType::Pointer pLinearSolver
+    ): mThisModelPart(rThisModelPart),
+       mpLinearSolver(pLinearSolver)
+{               
+    Parameters default_parameters = Parameters(R"(
+    {
+        "minimal_size"                        : 0.01,
+        "maximal_size"                        : 1.0,
+        "error"                               : 0.01,
+        "penalty_normal"                      : 1.0e4,
+        "penalty_tangential"                  : 1.0e4,
+        "echo_level"                          : 0,
+        "set_number_of_elements"              : false,
+        "number_of_elements"                  : 1000,
+        "average_nodal_h"                     : false
+    })" 
+    );
+    
+    ThisParameters.ValidateAndAssignDefaults(default_parameters);
+    
+    mMinSize = ThisParameters["minimal_size"].GetDouble();
+    mMaxSize = ThisParameters["maximal_size"].GetDouble();
+    mPenaltyNormal = ThisParameters["penalty_normal"].GetDouble();
+    mPenaltyTangent = ThisParameters["penalty_tangential"].GetDouble();
+    mEchoLevel = ThisParameters["echo_level"].GetInt();
+    mSetElementNumber = ThisParameters["set_number_of_elements"].GetBool();
+    mElementNumber = ThisParameters["number_of_elements"].GetInt();
+    mTargetError = ThisParameters["error"].GetDouble();
+    mAverageNodalH = ThisParameters["average_nodal_h"].GetBool();
+
+    // Creating a default solver in case none assigned
+    if (mpLinearSolver == nullptr) {
```

Comment:
This would be nicely solved if we manage to merge the linar solver factory in python.
for the time being i would throw an error telling that the solver needs to be passed from outside